package.json

```diff
@@ -92,23 +92,24 @@
     "!__tests__"
   ],
   "scripts": {
-    "docs": "npm run build && npm run docs:ci",
-    "docs:ci": "npm run docs:extract -- --local && npm run docs:generate",
-    "docs:generate": "api-documenter markdown --input-folder ./tmp --output-folder ./api/docs",
-    "docs:extract": "api-extractor run --verbose",
-    "build": "npm run clean && tsc",
-    "clean": "rm -rf dist",
-    "version:version-file": "genversion --template version-template.ejs --force lib/version.ts",
-    "version": "npm run version:version-file && npm run docs && git add ./lib/version.ts ./api/docs",
-    "test": "npm run test:ts && npm run test:legacy",
-    "test:ts": "jest",
-    "test:legacy": "npm run build -- --declaration false && ./test/scripts/vows.js test/*_test.js",
-    "typecheck": "tsc --noEmit",
-    "cover": "jest --coverage",
-    "lint": "eslint .",
-    "eslint": "eslint .",
-    "prettier": "prettier .",
-    "format": "npm run eslint -- --fix"
+    "build": "npm run _build:clean && npm run _build:compile",
+    "precommit": "npm run build && npm run _api:update && npm run _docs:generate && npm run _format:fix && npm run _lint:fix",
```

Comment:
I wasn't sure if `precommit` was a good name to use here or not as I wasn't intending it to be integrated with `git`. I was thinking more along the lines of "what tasks should you run before committing changes so that when you open a PR, none of the checks will fail for silly reasons"? 

Would you change your mind here if the script was renamed to `prepare-pr` instead? That way it overlaps less with `git` hooks terminology.

Reply:
`prepare-pr` would be fine.

If we want to do local checks, should we add git hooks? (In a follow up PR)

Reply:
Renamed to `prepare-pr`. I also changed that script to kick off a test run also so it can be used to both:
- ensure nothing was forgotten before opening a PR
- do a sanity check before publishing a new version

Reply:
Not sure how I feel about git hooks. I'm not entirely opposed but I find them to be more of a nuisance then a benefit. But maybe I've just been subjected to git hooks that try to do more validation than they should?